### ISSUE TITLE
gmic-2.1.5-r5 doesn't compile

### ISSUE THREAD

**Issue by ff2000** [stefantalpalaru/gentoo-overlay#19]
I already had the same issue with 2.0.0:
```
>>> Source configured.
>>> Compiling source in /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5 ...
>>> Working in BUILD_DIR: "/var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5_build"
ninja -v -j11 -l0
[1/8] cd /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5_build && /usr/bin/cmake -P /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5_build/gmic_stdlib_download.cmake && touch /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5/src/gmic_stdlib.h
[2/8] cd /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5_build && /usr/bin/cmake -P /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5_build/CImg_download.cmake && touch /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5/src/CImg.h
[3/8] /usr/lib64/distcc/bin/x86_64-pc-linux-gnu-g++  -I/usr/include/GraphicsMagick  -march=sandybridge -mno-aes -O2 -pipe -ggdb  -Dgmic_build -Dcimg_use_vt100 -Dgmic_is_parallel -Dcimg_use_abort  -std=gnu++11 -Wno-error=narrowing -fno-ipa-sra -fpermissive -fopenmp -Dcimg_use_openmp  -Dcimg_use_zlib  -Dcimg_display=1 -Dcimg_appname=\"gmic\"  -Dcimg_use_xshm  -Dcimg_use_fftw3  -fPIE    -Dcimg_use_magick  -Dcimg_use_tiff  -Dcimg_use_png  -Dcimg_use_jpeg  -Dgmic_dynamic_linking  -Dgmic_main -MD -MT CMakeFiles/gmic.dir/src/gmic.cpp.o -MF CMakeFiles/gmic.dir/src/gmic.cpp.o.d -o CMakeFiles/gmic.dir/src/gmic.cpp.o -c /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5/src/gmic.cpp
FAILED: CMakeFiles/gmic.dir/src/gmic.cpp.o 
/usr/lib64/distcc/bin/x86_64-pc-linux-gnu-g++  -I/usr/include/GraphicsMagick  -march=sandybridge -mno-aes -O2 -pipe -ggdb  -Dgmic_build -Dcimg_use_vt100 -Dgmic_is_parallel -Dcimg_use_abort  -std=gnu++11 -Wno-error=narrowing -fno-ipa-sra -fpermissive -fopenmp -Dcimg_use_openmp  -Dcimg_use_zlib  -Dcimg_display=1 -Dcimg_appname=\"gmic\"  -Dcimg_use_xshm  -Dcimg_use_fftw3  -fPIE    -Dcimg_use_magick  -Dcimg_use_tiff  -Dcimg_use_png  -Dcimg_use_jpeg  -Dgmic_dynamic_linking  -Dgmic_main -MD -MT CMakeFiles/gmic.dir/src/gmic.cpp.o -MF CMakeFiles/gmic.dir/src/gmic.cpp.o.d -o CMakeFiles/gmic.dir/src/gmic.cpp.o -c /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5/src/gmic.cpp
In file included from /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5/src/gmic.h:181:0,
                 from /var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5/src/gmic.cpp:2023:
/var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5/src/./CImg.h: In static member function ‘static double cimg_library::CImg<T>::_cimg_math_parser::mp_breakpoint(cimg_library::CImg<T>::_cimg_math_parser&)’:
/var/tmp/portage/media-gfx/gmic-2.1.5-r5/work/gmic-2.1.5/src/./CImg.h:20262:9: error: ‘cimg_abort_test’ was not declared in this scope
         cimg_abort_test;
         ^~~~~~~~~~~~~~~
```
and so on...

I tried to track it down but gmic is an absolute chaos of #defines, different repos and build systems. I'll try to build it locally from git now, but for "stable" releases compilation still should work...

I hope you can help me with this.
Thx!

**Comment by ff2000** [comment]
Hah! This seems to be an incompatibility with ninja. it now continued and is happily compiling the qt gui. And as I am writing this it successfully finished installation :)

You can fix this in the ebuild by adding `CMAKE_MAKEFILE_GENERATOR="emake"` before the inherit line.

**Comment by stefantalpalaru** [comment]
How did you get it to use ninja in the first place? You set that variable in make.conf?

**Comment by ff2000** [comment]
Yes, I set it in make.conf
ninja in general runs faster than make.
I created a gentoo bug: https://bugs.gentoo.org/643090
Which blocks the ninja-porting bug: https://bugs.gentoo.org/557992

[I switched it over to ninja quite some time back when webkit-gtk created an overhead of 30-40 minutes (!) with emake because the calculation for 2 target dependencies was extremely inefficient and the webkit-gtk-ebuild didn't force ninja at that time.]

**Comment by stefantalpalaru** [comment]
Fixed in gmic-2.1.8 and gmic-9999-r3.
